pkg/v3/plugin/coordinateblock.go

```diff
@@ -0,0 +1,92 @@
+package plugin
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg"
+	ocr2keepersv3 "github.com/smartcontractkit/ocr2keepers/pkg/v3"
+)
+
+type coordinateBlock struct {
+	threshold int
+	seen      map[heightHash]struct{}
+	recent    map[heightHash]int
+}
+
+func newCoordinateBlock(threshold int) *coordinateBlock {
+	return &coordinateBlock{
+		threshold: threshold,
+		seen:      make(map[heightHash]struct{}),
+		recent:    make(map[heightHash]int),
+	}
+}
+
+type heightHash struct {
+	number uint64
+	hash   string
+}
+
+func (p *coordinateBlock) add(observation ocr2keepersv3.AutomationObservation) {
+	rawHistory, ok := observation.Metadata[ocr2keepersv3.BlockHistoryObservationKey]
+	if !ok {
+		return
+	}
+
+	history, ok := rawHistory.(ocr2keepers.BlockHistory)
+	if !ok {
+		return
+	}
+
+	for _, key := range history.Keys() {
+		// TODO: use helper function for splitting keys
+		values := strings.Split(string(key), "|")
+		if len(values) != 2 {
+			continue
+		}
+
+		// TODO: don't use values at index
+		v, err := strconv.ParseUint(values[0], 10, 64)
+		if err != nil {
+			continue
+		}
+
+		// TODO: don't use values at index
+		height := heightHash{v, values[1]}
+
+		if _, present := p.seen[height]; !present {
+			p.seen[height] = struct{}{}
+			p.recent[height]++
```

Comment:
so we dont care about the value of `seen` map, it is just empty struct. And `recent` counts from 0